src/parser.rs

```diff
@@ -195,6 +234,7 @@ pub enum AstNode {
         field: NodeId,
     },
     Block(BlockId),
+    Pipeline(Pipeline),
```

Comment:
It would be better to put the pipelines into a vector in the Compiler and refer to them by PipelineId, similar to how blocks are addressed by BlockId. This it to avoid allocating the pipelines at runtime because we could preallocate the Compiler vectors with some reasonable size in advance.

Reply:
Sure!  I moved it into `Compiler`